### ISSUE TITLE
Oracle: RF01 should not trigger when calling sequence pseudocollums like NEXTVAL or CURRVAL

### ISSUE THREAD

**Issue by 123Haynes** [sqlfluff/sqlfluff#7634]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm trying to lint the following SQL script with the oracle dialect:

`SELECT emloye_seq.NEXTVAL, name, a, b FROM Employee;`

This currently triggers the rule RF01.

### Expected Behaviour

Ideally the rule should not trigger for sequence pseudocollums like NEXTVAL or CURRVAL.

### Observed Behaviour

```
FAIL
L:   2 | P:   5 | RF01 | Reference 'emloye_seq.nextval' refers to table/view not
                       | found in the FROM clause or found in ancestor statement.
                       | [references.from]
```

### How to reproduce

call `sqlfluff lint --dialect oracle file.sql` on the SQL mentioned above.

### Dialect

oracle

### Version

4.0.4

### Configuration

nothing apart from the oracle dialect.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
